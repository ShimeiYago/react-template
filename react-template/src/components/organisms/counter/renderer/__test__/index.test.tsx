import React from 'react';
import { create } from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import { Renderer, Props } from '..';

const basicProps: Props = {
  count: 0,
  counterStatus: 'idle',

  decrement: jest.fn(),
  increment: jest.fn(),
  incrementByAmount: jest.fn(),
  incrementIfOdd: jest.fn(),
  fetchCount: jest.fn(),
  postCount: jest.fn(),
}

describe('Shallow Snapshot Tests', () => {
  const renderer = ShallowRenderer.createRenderer();

  it('basic', () => {
    const result = renderer.render(<Renderer {...basicProps} />);
    expect(result).toMatchSnapshot();
  });

  it('loading', () => {
    const result = renderer.render(
      <Renderer {...basicProps} counterStatus='loading' />
    );
    expect(result).toMatchSnapshot();
  });

  it('failed', () => {
    const result = renderer.render(
      <Renderer {...basicProps} counterStatus='failed' />
    );
    expect(result).toMatchSnapshot();
  });
});

test('incrementAmount should change when value is inputted.', () => {
  const inputValue = 'input value';

  const testInstance = create(<Renderer {...basicProps} />).root;
  const input = testInstance.findByType('input');

  expect(testInstance.instance.state.incrementAmount).toBe('2');
  input.props.onChange({target:{value:inputValue}});
  expect(testInstance.instance.state.incrementAmount).toBe(inputValue);
});
