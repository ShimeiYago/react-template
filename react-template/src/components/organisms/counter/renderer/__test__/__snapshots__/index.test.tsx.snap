// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<div>
  <div
    className="row"
  >
    <button
      aria-label="Decrement value"
      className="button"
      onClick={[Function]}
    >
      -
    </button>
    <span
      className="value"
    >
      0
    </span>
    <button
      aria-label="Increment value"
      className="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <input
      aria-label="Set increment amount"
      className="textbox"
      onChange={[Function]}
      value="2"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Add Amount
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Add If Odd
    </button>
    <button
      className="asyncButton"
      disabled={false}
      onClick={[Function]}
    >
      Fetch
    </button>
    <button
      className="asyncButton"
      disabled={false}
      onClick={[Function]}
    >
      Post
    </button>
  </div>
</div>
`;

exports[`Shallow Snapshot Tests loading 1`] = `
<div>
  <div
    className="row"
  >
    <button
      aria-label="Decrement value"
      className="button"
      onClick={[Function]}
    >
      -
    </button>
    <span
      className="value"
    >
      0
    </span>
    <button
      aria-label="Increment value"
      className="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <input
      aria-label="Set increment amount"
      className="textbox"
      onChange={[Function]}
      value="2"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Add Amount
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Add If Odd
    </button>
    <button
      className="asyncButton"
      disabled={true}
      onClick={[Function]}
    >
      Fetch
    </button>
    <button
      className="asyncButton"
      disabled={true}
      onClick={[Function]}
    >
      Post
    </button>
  </div>
</div>
`;

exports[`Shallow Snapshot Tests with error 1`] = `
<div>
  <div>
    error
  </div>
  <div
    className="row"
  >
    <button
      aria-label="Decrement value"
      className="button"
      onClick={[Function]}
    >
      -
    </button>
    <span
      className="value"
    >
      0
    </span>
    <button
      aria-label="Increment value"
      className="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <input
      aria-label="Set increment amount"
      className="textbox"
      onChange={[Function]}
      value="2"
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Add Amount
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Add If Odd
    </button>
    <button
      className="asyncButton"
      disabled={false}
      onClick={[Function]}
    >
      Fetch
    </button>
    <button
      className="asyncButton"
      disabled={false}
      onClick={[Function]}
    >
      Post
    </button>
  </div>
</div>
`;
